### ISSUE TITLE
Improve System.Reflection.Context.Tests code coverage

### ISSUE THREAD

**Issue by ViktorHofer** [dotnet/corefx#30418]
Currently at ~35% line coverage.

To test and measure coverage run `msbuild /t:RebuildAndTest /p:Coverage=true /v:m` inside the test project `corefx/src/System.Reflection.Context/tests/`.

**Comment by mbrameld** [comment]
I'd like to take this one on and add some tests. I'm following [this guide to build and run the tests](https://github.com/dotnet/corefx/wiki/Build-and-run-tests). Everything works fine from the command line, but when I try to run the tests in Visual Studio 2017 by setting the test project as startup and running it, I get an error saying a project with an output type of class library can't be started. The tests all show up in the test explorer, but when I try to run them it tells me: No test is available in corefx\bin\AnyOS.AnyCPU.Debug\System.Reflection.Context.Tests\netcoreapp\System.Reflection.Context.Tests.dll. Make sure that test discoverer & executors are registered and platform & framework version settings are appropriate and try again.

Am I missing a step somewhere?

**Comment by ViktorHofer** [comment]
Debugging is currently broken in VS: https://github.com/dotnet/corefx/issues/30913

But should be back in a few days (finger crossed).

**Comment by mbrameld** [comment]
Ah, thanks! I'll keep an eye on that issue and come back to this one.

**Comment by ViktorHofer** [comment]
Yeah, sure. Thanks a lot for the help! I'll keep you updated. But you don't really need VS. The easiest way to test is:

```
cd src/System.Reflection.Context/tests
msbuild /t:RebuildAndTest /p:Coverage=true
```

Make sure you run that inside the VS Developer Command Prompt 2017.

**Comment by mbrameld** [comment]
That does run the tests for me, but I could benefit from stepping through things. I'm not familiar with how System.Reflection.Context all works yet and I'm struggling a little without the debugger!

**Comment by ViktorHofer** [comment]
Understandable. I suggest to wait a few days then 👍

**Comment by danmosemsft** [comment]
Or add `while(!Debugger.IsAttached) Thread.Sleep(1000)` to a test them attach VS at leisure?

**Comment by ViktorHofer** [comment]
> Or add while(!Debugger.IsAttached) Thread.Sleep(1000) to a test them attach VS at leisure?

Or simply a `Debugger.Break();`

**Comment by mbrameld** [comment]
I accepted it, thank you!

**Comment by ViktorHofer** [comment]
I flagged it as easy but it could me more of an intermediate level of work. If you have questions / need help, reach out :)

**Comment by Adam25T** [comment]
Hello, in the coming week a group of people trying to gain experience with open source are going to be contributing to corefx. Issues that focus on increasing test coverage seem like a good place to start. How do I become a collaborator on this repo?

**Comment by ViktorHofer** [comment]
Hi @Adam25T. Awesome, great to hear!! Few things:

1. If you tell me all the github handles of people who are going to contribute I can add them now. Please make sure to unsubscribe from all the repos as being a members means auto-subscribing to 50+ repos which results in ~500 emails a day.
2. If you are looking for issue that are relatively easy for first-time-contributors, not just code-coverage, then I can make sure to label all appropriate issues till next week with the `easy` and `up-for-grabs` label. If your group is part of a Hackathon we can probably also support you in our Gitter channel to unblock you and make sure that you can work in a productive way. Our docs should be self-explanatory but some advanced topics like System.Private.CoreLib work in coreclr requires tend to cause troubles.

cc @karelz @danmosemsft

**Comment by Adam25T** [comment]
Thanks for the prompt reply @ViktorHofer . Technically it is a small scale hackathon for junior level people and I don't have their github handles. Just looking for some simple issues to try and tackle.

**Comment by karelz** [comment]
@Adam25T check our "Pick issue" guide in developer docs: https://github.com/dotnet/corefx/wiki/Pick-issue
